src/CoreWCF.Primitives/tests/DependencyInjection/ServiceInstanceContextModeTests.cs

```diff
@@ -0,0 +1,174 @@
+﻿using CoreWCF.Channels;
+using CoreWCF.Description;
+using CoreWCF.Primitives.Tests.Helpers;
+using Helpers;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using Xunit;
+
+namespace DependencyInjection
+{
+    public class ServiceInstanceContextModeTests
+    {
+        [Fact]
+        public static void InstanceContextMode_Single()
+        {
+            SingleInstanceContextSimpleService.CreationCount = 0;
+            SingleInstanceContextSimpleService.DisposalCount = 0;
+            var services = new ServiceCollection();
+            var serviceInstance = new SingleInstanceContextSimpleService();
+            services.AddSingleton(serviceInstance);
+            TestHelper.BuildDispatcherAndCallService<SingleInstanceContextSimpleService>(services);
```

Comment:
Did you consider making 2 calls. so that the CreateCount would be 2 if it created multiple instances but still skipped Dispose ?

Reply:
Good point. I'll add that. I was concentrating on making sure that the injected instance is the only instance created and making a single request is sufficient to show this. I think you can safely infer that it's doing the right thing with the existing code but explicit is better then inference if you can test something explicitly so I'll make the change.